src/maybe.ts

```diff
@@ -105,10 +105,10 @@ export class Maybe<T> implements Monad<T>, Functor<T>, Eq<Maybe<T>> {
      * @returns {Maybe<T>} A Maybe object containing the value passed in input. If t is null
      *     or undefined, the Maybe object is filled with Nothing.
      */
-    static maybe<T>(t: T) {
-        return t === null || t === undefined ?
-            new Maybe<T>(MaybeType.Nothing) :
-            new Maybe<T>(MaybeType.Just, t);
+    static maybe<T>(t?: T): Maybe<T> {
```

Comment:
You'll probably want to change `T` to `T | null` since `strictNullChecks` removes `null` from the domain of every type.